### PR TITLE
Ns/refactor zk tests

### PR DESCRIPTION
**Pull request by nsarlin-zama**
Removes code duplication in zk tests

### CODE REVIEW

File: tfhe-zk-pok/src/proofs/pke_v2.rs [hunk 1]
```diff
@@ -2357,165 +2357,64 @@ pub fn verify<G: Curve>(
...
         let fake_e1 = (0..d)
-            .map(|_| (rng.gen::<u64>() % (2 * B)) as i64 - B as i64)
-            .collect::<Vec<_>>();
-        let e2 = (0..k)
-            .map(|_| (rng.gen::<u64>() % (2 * B)) as i64 - B as i64)
+            .map(|_| (rng.gen::<u64>() % (2 * b_i)) as i64 - b_i as i64)
             .collect::<Vec<_>>();
         let fake_e2 = (0..k)
-            .map(|_| (rng.gen::<u64>() % (2 * B)) as i64 - B as i64)
+            .map(|_| (rng.gen::<u64>() % (2 * b_i)) as i64 - b_i as i64)
```
Comment (IceTDrinker) [inline tfhe-zk-pok/src/proofs/pke_v2.rs:2393]: are we sure about the B vs b_i ?

File: tfhe-zk-pok/src/proofs/pke.rs [hunk 2]
```diff
@@ -1678,123 +1478,30 @@ mod tests {
     fn test_proof_compression() {
         let d = 2048;
         let k = 320;
-        let big_b = 1048576;
+        let b_i = 512;
```
Comment (IceTDrinker) [inline tfhe-zk-pok/src/proofs/pke.rs:1481]: why the change ?

Comment (nsarlin-zama) [reply tfhe-zk-pok/src/proofs/pke.rs:1481]: I think it was a mistake in the previous version, since the other tests for pke v1 use `b_i = 512`

File: tfhe-zk-pok/src/proofs/pke_v2.rs (hunk 1 above)
Comment (nsarlin-zama) [reply tfhe-zk-pok/src/proofs/pke_v2.rs:2393]: in both proofs they play the same role (noise bound), so I think we should give them the same name in pke and pke_v2. Before the pr this is called B in pke and b_i in pke_v2.

Comment (IceTDrinker) [reply tfhe-zk-pok/src/proofs/pke_v2.rs:2393]: I think it's linked to the notations in the technical notes, they are not strictly the same IIRC as we need to derive an upper bound of the std dev in v2

Comment (nsarlin-zama) [reply tfhe-zk-pok/src/proofs/pke_v2.rs]: I used B everywhere and used the value from the TUniform params

File: tfhe-zk-pok/src/proofs/mod.rs [hunk 3]
```diff
@@ -174,3 +174,202 @@ pub mod pke;
 pub mod pke_v2;
 pub mod range;
 pub mod rlwe;
+
+#[cfg(test)]
+mod test {
+    #![allow(non_snake_case)]
```
Comment (IceTDrinker) [inline tfhe-zk-pok/src/proofs/mod.rs:180]: what for ?

File: tfhe-zk-pok/src/proofs/mod.rs [hunk 4]
```diff
@@ -174,3 +174,202 @@ pub mod pke;
...
+        pub(super) d: usize,
+        pub(super) k: usize,
+        pub(super) B: u64,
+        pub(super) q: u64,
+        pub(super) t: u64,
+        pub(super) msbs_zero_padding_bit_count: u64,
+    }
+
+    pub(super) const DEFAULT_TEST_PKE_PARAMS: PkeTestParameters = PkeTestParameters {
+        d: 2048,
```
Comment (IceTDrinker) [inline tfhe-zk-pok/src/proofs/mod.rs:246]: not the correct d, you want the one from tfhe/src/shortint/parameters/compact_public_key_only/p_fail_2_minus_64/ks_pbs.rs:9

File: tfhe-zk-pok/src/proofs/mod.rs [hunk 5]
```diff
@@ -174,3 +174,202 @@ pub mod pke;
...
+        k: 320,
+        B: 131072, // 2^17, taken from TUniform 2^-64 parameters
+        q: 0,
+        t: 1024,
```
Comment (IceTDrinker) [inline tfhe-zk-pok/src/proofs/mod.rs:250]: this t is not correct either if we want BC like params

File: tfhe-zk-pok/src/proofs/pke_v2.rs [hunk 6]
```diff
@@ -2758,119 +2579,28 @@ mod tests {
...
-        let effective_cleartext_t = t >> msbs_zero_padding_bit_count;
-
-        let delta = {
-            let q = if q == 0 { 1i128 << 64 } else { q as i128 };
-            (q / t as i128) as u64
-        };
+        let PkeTestParameters {
+            d,
+            k,
+            B,
```
Comment (IceTDrinker) [inline tfhe-zk-pok/src/proofs/pke_v2.rs:2585]: we don't recompute the upper bound of the std dev with the formula I gave you for TUniform stuff ?

File: tfhe-zk-pok/src/proofs/mod.rs (hunk 3 above)
Comment (nsarlin-zama) [reply tfhe-zk-pok/src/proofs/mod.rs:180]: because of the var named 'B'

File: tfhe-zk-pok/src/proofs/pke_v2.rs (hunk 6 above)
Comment (nsarlin-zama) [reply tfhe-zk-pok/src/proofs/pke_v2.rs:2585]: If I understand correctly the formula is to convert from a gaussian stddev, but here we already have a TUniform value

Comment (IceTDrinker) [reply tfhe-zk-pok/src/proofs/pke_v2.rs:2585]: actually the fomula should give an equivalent std dev for the TUniform that can safely be used in places where a gaussian is expected, not sure it is the exact std dev for a TUniform distribution however, guess we could check that formula

File: tfhe-zk-pok/src/proofs/pke.rs [hunk 7]
```diff
@@ -1457,56 +1369,30 @@ mod tests {
...
+            r,
+            m: _m,
+            b,
+            c1,
+            c2,
+            metadata,
+        } = PkeTestProofInputs::gen(rng, PKEV1_TEST_PARAMS);
 
         // Generate messages with padding set to fail verification
         let m = {
```
Comment (IceTDrinker) [inline tfhe-zk-pok/src/proofs/pke.rs:1398]: this is not correct, here this m is equivalent to a "fake m" what we want for the padding bit test is :

generate a legit encryption but with m that has at least one slot with a padding bit set and with that proven data we will check that the verification fails, indicating we are able to properly detect when a padding bit is set

File: tfhe-zk-pok/src/proofs/pke_v2.rs [hunk 8]
```diff
@@ -2545,41 +2464,30 @@ mod tests {
...
+            r,
+            m: _m,
+            b,
+            c1,
+            c2,
+            metadata,
+        } = PkeTestProofInputs::gen(rng, PKEV2_TEST_PARAMS);
 
         // Generate messages with padding set to fail verification
         let m = {
```
Comment (IceTDrinker) [inline tfhe-zk-pok/src/proofs/pke_v2.rs:2493]: same thing for the padding bit test, this is a fake m test

**Review by IceTDrinker — CHANGES_REQUESTED** [review]
Some tests are not equivalent and we need to be rock solid on the padding bit thing :) otherwise looks good !

File: tfhe-zk-pok/src/proofs/pke.rs (hunk 7 above)
Comment (nsarlin-zama) [reply tfhe-zk-pok/src/proofs/pke.rs:1398]: You're right, good catch !

**Review by IceTDrinker — APPROVED** [review]
